src/features/creators/creationHelpers.ts

```diff
@@ -0,0 +1,199 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { extensions, l10n, QuickInputButtons, Uri, window } from 'vscode';
+import { CreateEnvironmentOptions } from '../../api';
+import { traceError, traceVerbose } from '../../common/logging';
+import { showQuickPickWithButtons } from '../../common/window.apis';
+import { EnvironmentManagers, InternalEnvironmentManager } from '../../internal.api';
+
+/**
+ * Prompts the user to choose whether to create a new virtual environment (venv) for a project, with a clearer return and early exit.
+ * @returns {Promise<boolean | undefined>} Resolves to true if 'Yes' is selected, false if 'No', or undefined if cancelled.
+ */
+export async function promptForVenv(callback: () => void): Promise<boolean | undefined> {
+    try {
+        const venvChoice = await showQuickPickWithButtons([{ label: l10n.t('Yes') }, { label: l10n.t('No') }], {
```

Comment:
We have localized versions of this in `localize.ts`. You can address this later.